src/main/controllers/features/unincorporated/unincorporatedWhatIsYourRoleController.ts

```diff
@@ -36,7 +40,9 @@ export const post = async (req: Request, res: Response, next: NextFunction) => {
                 previousPage: addLangToUrl(BASE_URL + UNINCORPORATED_WHAT_IS_THE_BUSINESS_NAME, lang),
                 currentUrl: BASE_URL + UNINCORPORATED_WHAT_IS_YOUR_ROLE,
                 pageProperties: pageProperties,
-                payload: req.body
+                payload: req.body,
+                acspType: acspType,
+                unincorporatedBusinessName: businessName
             });
         } else {
             if (req.body.WhatIsYourRole === "SOMEONE_ELSE") {
```

Comment:
Make the decision in a function and send the redirect url back.
I would have liked this bit of logic as well in service layer, but that may be too expensive. So we can create the function here itself and calculate the return url.

Reply:
I haven't made any change here

Reply:
But could you please make it a function in the controller which makes this decision.

Reply:
Done